extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/ConsumeProcessor.java

```diff
@@ -187,6 +212,28 @@ void generateConsumeRoutes(
         }
     }
 
+    private String findEndpointMethodName(final String propertyName, ClassInfo declaringClass, List<String> triedMethods) {
+        /* Here we attempt to mimic what Camel does
+         * in org.apache.camel.impl.engine.CamelPostProcessorHelper.doGetEndpointInjection(Object, String, String) */
+        final String isGetter = "is" + StringHelper.capitalize(propertyName, false);
+        final String getGetter = "get" + StringHelper.capitalize(propertyName, false);
+        Optional<MethodInfo> method = Stream.of(isGetter, getGetter, isGetter + "Endpoint", getGetter + "Endpoint")
+                .peek(triedMethods::add)
+                .map(declaringClass::method)
+                .filter(m -> m != null)
+                .findFirst();
+        if (propertyName.startsWith("on")) {
```

Comment:
So, it can be that we have found a matching method but still we try onXXX ?

Reply:
Good catch, thanks! Let me fix it.